docs/manual-configuration.md

```diff
@@ -0,0 +1,135 @@
+
+![A three stage pipeline that starts with a list of Java sources, creates a list of SemanticDB files that then become a single LSIF index.](docs/img/semanticdb-javac-pipeline.svg)
+
+## Step 1: Add SemanticDB compiler plugin to the classpath
+
+The SemanticDB compiler plugin is published to Maven Central as a
+zero-dependency Java library that you can install like any normal dependency.
+
+[![](https://img.shields.io/maven-central/v/com.sourcegraph/semanticdb-javac)](https://repo1.maven.org/maven2/com/sourcegraph/semanticdb-javac/)
+
+If you're using Gradle.
+
+```groovy
+implementation group: 'com.sourcegraph', name: 'semanticdb-javac', version: '@STABLE_VERSION@'
+```
+
+If you're using Maven.
+
+```xml
+<dependency>
+    <groupId>com.sourcegraph</groupId>
+    <artifactId>semanticdb-javac</artifactId>
+    <version>@STABLE_VERSION@</version>
+</dependency>
+```
+
+If you're using sbt.
+
+```scala
+libraryDependencies += "com.sourcegraph" % "semanticdb-javac" % "@STABLE_VERSION@"
+```
+
+## Step 2: enable `-Xplugin:semanticdb` compiler option
+
+Add `-Xplugin:semanticdb` to your compiler options to enable the SemanticDB
+compiler plugin. To do this you need to explicitly configure two directories:
+
+- `-sourceroot:PATH`: the absolute path to the root directory of your codebase.
+  It's important that all of the source files that should be index live under
+  this directory.
+- `-targetroot:PATH`: the absolute path to the directory where to generate
+  SemanticDB file. This directory can be anywhere on your file system.
+
+If you're using Gradle.
+
+```groovy
+tasks.withType(JavaCompile) {
+    def sourceroot = rootProject.projectDir
+    def targetroot = new File(rootProject.buildDir, "semanticdb-targetroot")
+    options.compilerArgs << '-Xplugin:semanticdb -sourceroot:$sourceroot -targetroot:$targetroot'
+}
+```
+
+If you're using Maven.
+
+```diff
+ <project>
+   [...]
+   <build>
+     [...]
+     <plugins>
+       <plugin>
+         <groupId>org.apache.maven.plugins</groupId>
+         <artifactId>maven-compiler-plugin</artifactId>
+         <version>3.8.1</version>
+         <configuration>
+           <compilerArgs>
++            <arg>-Xplugin:semanticdb -sourceroot:${session.executionRootDirectory} -targetroot:${session.executionRootDirectory}/target/semanticdb-targetroot</arg>
+           </compilerArgs>
+         </configuration>
+       </plugin>
+     </plugins>
+     [...]
+   </build>
+   [...]
+ </project>
+```
+
+If you're using sbt.
+
+```scala
+Compile / javacOptions += {
+    val sourceroot = (ThisBuild / baseDirectory).value
+    val targetroot = (ThisBuild / target).value / "semanticdb-targetroot"
+    s"-Xplugin:semanticdb -sourceroot:$sourceroot -targeroot:$sourceroot"
+}
+```
+
+## Step 4: compile the codebase
+
+Compile all source files in the codebase once the compiler setting has been
+configured. The exact command depends on your build tool. Below are some
+examples:
+
+- Gradle: `./gradlew clean compileTestJava`
```

Comment:
I know `compileTestJava` technically invokes `compileJava` as a dependency, so unless explicitly invoking both would result in indexing twice, I'd personally also include `compileJava` so users mightnt be left wondering why we're only suggesting to run `compileTestJava` as an example :smile: 

```suggestion
- Gradle: `./gradlew clean compileJava compileTestJava`
```